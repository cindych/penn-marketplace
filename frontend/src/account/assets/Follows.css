.flex {
  display:flex;
  align-items: center;
  flex-direction: row;
}

.pagination-bar {
  justify-content: right;
  flex-grow: 5;
}

.box {
  background-color: white;
  border: 1px solid black;
}

.table-row {
  display: grid;
  grid-template-columns: 4fr 1fr 1fr;
  white-space: nowrap;
}

.table-item {
  text-align: center;
}

button {
  border: none;
  background: none;
}