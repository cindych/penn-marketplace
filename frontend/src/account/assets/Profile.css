#dark-mode-btn {
  background: none;
  border: none;
  padding-left: 0px;
  padding-top: 0px;
}

.flex {
  display:flex;
  align-items: center;
  flex-direction: row;
}

#reports {
  justify-content: right;
  flex-grow: 5;
}

#profile {
  border: 1px solid black;
  background: white;
}

p {
  width: 25%;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  height: 19px;
  letter-spacing: 0em;
  text-align: left;
}