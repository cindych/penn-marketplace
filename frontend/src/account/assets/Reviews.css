.rating {
  display: inline-flex;
}

#reviews {
  margin-top: 3%;
  background-color: white;
  border: 1px solid black;
}

.review {
  background-color: #EFF0F0;
  margin-top: 1%;
  margin-bottom: 1%;
  border-radius: 5px;
}