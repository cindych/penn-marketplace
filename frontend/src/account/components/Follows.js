import React, { useState } from 'react';
import axios from 'axios';
import '../assets/Follows.css';
import NextIcon from '../assets/Next.png';
import BackIcon from '../assets/Back.png';
import ReviewIcon from '../assets/Review.png';
import UnfollowIcon from '../assets/Unfollow.png';

const Follows = ({ followersProp, following }) => {
  const [followedUsers, setFollowedUsers] = useState(following);
  const [followers, setFollowers] = useState(followersProp);
  const [followedUsersPage, setFollowedUsersPage] = useState(1);
  const [followersPage, setFollowersPage] = useState(1);
  const followedUsersItems = followedUsers.slice(
    (followedUsersPage - 1) * 10,
    followedUsersPage * 10,
  );
  const followersItems = followers.slice(
    (followersPage - 1) * 10,
    followersPage * 10,
  );

  function prevFollowedUsersPage() {
    if (followedUsersPage > 1) {
      setFollowedUsersPage(followedUsersPage - 1);
    }
  }

  function nextFollowedUsersPage() {
    if (followedUsersPage * 10 < followedUsers.length) {
      setFollowedUsersPage(followedUsersPage + 1);
    }
  }

  function prevFollowersPage() {
    if (followersPage > 1) {
      setFollowersPage(followersPage - 1);
    }
  }

  function nextFollowersPage() {
    if (followersPage * 10 < followers.length) {
      setFollowersPage(followersPage + 1);
    }
  }

  async function unfollow(e) {
    let indexToDelete;
    if (e.target.tagName === 'IMG') {
      indexToDelete = e.target.parentNode.value;
    } else {
      indexToDelete = e.target.value;
    }
    const newFollowedUsers = [...followedUsers];
    const removedFollowing = newFollowedUsers.splice(indexToDelete, 1);
    try {
      await axios.post('/account/unfollow', { removedFollowing: removedFollowing[0], newFollowList: newFollowedUsers });
      setFollowedUsers(newFollowedUsers);
      if (newFollowedUsers.length < followedUsersPage * 10 - 9 && followedUsersPage > 1) {
        setFollowedUsersPage(followedUsersPage - 1);
      }
    } catch (error) {
      throw new Error('Error unfollowing user');
    }
  }

  async function removeFollower(e) {
    let indexToDelete;
    if (e.target.tagName === 'IMG') {
      indexToDelete = e.target.parentNode.value;
    } else {
      indexToDelete = e.target.value;
    }
    const newFollowers = [...followers];
    const removedFollowing = newFollowers.splice(indexToDelete, 1);
    try {
      await axios.post('/account/unfollow', { removedFollowing: removedFollowing[0], newFollowList: newFollowers });
      setFollowers(newFollowers);
      if (newFollowers.length < followersPage * 10 - 9 && followersPage > 1) {
        setFollowersPage(followersPage - 1);
      }
    } catch (error) {
      throw new Error('Error unfollowing user');
    }
  }

  return (
    <div style={{ width: '100%' }}>
      <div>
        <div className="flex">
          <h1>Followed</h1>
          {followedUsers.length === 0 ? null
            : (
              <div className="flex pagination-bar">
                <button type="button" onClick={prevFollowedUsersPage}>
                  <img src={BackIcon} alt="Back Arrow Icon" />
                </button>
                &nbsp;{(followedUsersPage - 1) * 10 + 1} -&nbsp;
                {followedUsersPage * 10 > followedUsers.length
                  ? followedUsers.length : followedUsersPage * 10}&nbsp;
                of {followedUsers.length}&nbsp;
                <button type="button" onClick={nextFollowedUsersPage}>
                  <img src={NextIcon} alt="Next Arrow Icon" />
                </button>
              </div>
            )}
        </div>
        {followedUsers.length === 0 ? <div>You are not following anyone.</div>
          : (
            <div className="box">
              <div style={{ padding: '1% 2%' }}>
                <div className="table-row">
                  <br />
                  <div className="table-item">
                    Review
                  </div>
                  <div className="table-item">
                    Unfollow
                  </div>
                </div>
                {followedUsersItems.map((followedUser, index) => (
                  <div key={followedUser.followingEmail} className="table-row">
                    <p>
                      {followedUser.followingName}
                    </p>
                    <div className="table-item">
                      <img src={ReviewIcon} alt="Review icon" />
                    </div>
                    <div className="table-item">
                      <button type="button" onClick={unfollow} value={index + (followedUsersPage - 1) * 10}>
                        <img src={UnfollowIcon} alt="Unfollow icon" />
                      </button>
                    </div>
                  </div>
                ))}
              </div>
            </div>
          )}
      </div>
      <div style={{ marginTop: '3%' }}>
        <div className="flex">
          <h1>Followers</h1>
          {followers.length === 0 ? null
            : (
              <div className="flex pagination-bar">
                <button type="button" onClick={prevFollowersPage}>
                  <img src={BackIcon} alt="Back Arrow Icon" />
                </button>
                {followersPage * 10 > followers.length
                  ? followers.length : followersPage * 10}&nbsp;
                of {followers.length}&nbsp;
                <button type="button" onClick={nextFollowersPage}>
                  <img src={NextIcon} alt="Next Arrow Icon" />
                </button>
              </div>
            )}
        </div>
        {followers.length === 0 ? <div>You have no followers.</div>
          : (
            <div className="box">
              <div style={{ padding: '1% 2%' }}>
                <div className="table-row">
                  <br />
                  <div>
                    <br />
                  </div>
                  <div className="table-item">
                    Remove
                  </div>
                </div>
                {followersItems.map((follower, index) => (
                  <div key={follower.followerEmail} className="table-row">
                    <p>
                      {follower.followerName}
                    </p>
                    <div>
                      <br />
                    </div>
                    <div className="table-item">
                      <button type="button" onClick={removeFollower} value={index + (followersPage - 1) * 10}>
                        <img src={UnfollowIcon} alt="Remove icon" />
                      </button>
                    </div>
                  </div>
                ))}
              </div>
            </div>
          )}
      </div>
    </div>
  );
};

export default Follows;
