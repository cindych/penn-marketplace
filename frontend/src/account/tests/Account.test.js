/**
* @jest-environment jsdom
*/
/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import axios from 'axios';
import userEvent from '@testing-library/user-event';
import MockAdapter from 'axios-mock-adapter';
import Profile from '../components/Profile';
import Reviews from '../components/Reviews';
import Follows from '../components/Follows';
import Blocked from '../components/Blocked';
import SearchUsers from '../components/SearchUsers';

let testUser;
let mockAxios;

beforeEach(() => {
  mockAxios = new MockAdapter(axios);
  testUser = {
    name: 'John Doe',
    email: 'jdoe@sas.upenn.edu',
    reviews: [
      {
        authorEmail: 'cevans@sas.upenn.edu',
        authorName: 'Cadel Evans',
        recipient: 'jdoe@sas.upenn.edu',
        reviewContent: 'Good experience with this seller.',
        reviewRating: 5,
      },
      {
        authorEmail: 'fschleck@sas.upenn.edu',
        authorName: 'Frank Schleck',
        recipient: 'jdoe@sas.upenn.edu',
        reviewContent: 'The chair I bought had a missing leg.',
        reviewRating: 2,
      },
    ],
    followers: [
      {
        followerName: 'Cadel Evans',
        followerEmail: 'cevans@sas.upenn.edu',
        followingName: 'John Doe',
        followingEmail: 'jdoe@sas.upenn.edu',
      },
      {
        followerName: 'Frank Schleck',
        followerEmail: 'fschleck@sas.upenn.edu',
        followingName: 'John Doe',
        followingEmail: 'jdoe@sas.upenn.edu',
      },
      {
        followerName: 'Alex Dowsett',
        followerEmail: 'adowsett@sas.upenn.edu',
        followingName: 'John Doe',
        followingEmail: 'jdoe@sas.upenn.edu',
      },
    ],
    following: [
      {
        followerName: 'John Doe',
        followerEmail: 'jdoe@sas.upenn.edu',
        followingName: 'Cadel Evans',
        followingEmail: 'cevans@sas.upenn.edu',
      },
      {
        followerName: 'John Doe',
        followerEmail: 'jdoe@sas.upenn.edu',
        followingName: 'Alberto Contador',
        followingEmail: 'acontador@sas.upenn.edu',
      },
      {
        followerName: 'John Doe',
        followerEmail: 'jdoe@sas.upenn.edu',
        followingName: 'Levi Leipheimer',
        followingEmail: 'leipheimer@sas.upenn.edu',
      },
    ],
    blocked: [
      {
        blockerName: 'John Doe',
        blockerEmail: 'jdoe@sas.upenn.edu',
        blockedUserName: 'Levi Leipheimer',
        blockedUserEmail: 'leipheimer@sas.upenn.edu',
      },
    ],
    reports: [
      {
        authorEmail: 'csastre@sas.upenn.edu',
        recipientEmail: 'jdoe@sas.upenn.edu',
        reportContent: 'Never sent any of the items I bought.',
      },
    ],
  };
});

describe('UI testing for the Profile component', () => {
  test('Test 1: Snapshot on Profile Screen for user with a report', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Profile user={testUser} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Test 2: Snapshot on Profile screen for user with no reports', () => {
    testUser.reports = [];
    const component = renderer.create(
      <BrowserRouter>
        <Profile user={testUser} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('UI testing for the reviews component', () => {
  test('Test 1: Snapshot on Review Screen for user with reviews', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Reviews reviews={testUser.reviews} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    render(<BrowserRouter><Reviews reviews={testUser.reviews} /></BrowserRouter>);
    const review = screen.getByText(/The chair I bought had a missing leg./);
    expect(review).toBeInTheDocument();
    const reviewTwo = screen.getByText(/Good experience with this seller./);
    expect(reviewTwo).toBeInTheDocument();
    const averageRating = screen.getByText(/Reviews - Average Rating: 3.5/);
    expect(averageRating).toBeInTheDocument();
  });

  test('Test 2: Snapshot on Review screen for user with no reviews', () => {
    testUser.reviews = [];
    const component = renderer.create(
      <BrowserRouter>
        <Reviews reviews={testUser.reviews} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    render(<BrowserRouter><Reviews reviews={testUser.reviews} /></BrowserRouter>);
    const review = screen.getByText(/You have no reviews./);
    expect(review).toBeInTheDocument();
  });
});

describe('UI testing for the follows component', () => {
  test('Test 1: Snapshot on Review screen for user with followers and followed users', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Follows userProfile={testUser} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Test 2: User with no followers', () => {
    testUser.followers = [];
    render(
      <BrowserRouter>
        <Follows userProfile={testUser} />
      </BrowserRouter>,
    );
    const text = screen.getByText(/You have no followers./);
    expect(text).toBeInTheDocument();
  });
  test('Test 3: User who is not following anyone', () => {
    testUser.following = [];
    render(
      <BrowserRouter>
        <Follows userProfile={testUser} />
      </BrowserRouter>,
    );
    const text = screen.getByText(/You are not following anyone./);
    expect(text).toBeInTheDocument();
  });
  test('Test 4: User unfollows someone.', async () => {
    render(
      <BrowserRouter>
        <Follows userProfile={testUser} />
      </BrowserRouter>,
    );
    const followedUsersCopy = [...testUser.following];
    const removedFollowing = followedUsersCopy.splice(0, 1);
    mockAxios.onPost('/account/unfollow', { removedFollowing: removedFollowing[0], newFollowList: followedUsersCopy })
      .reply(200);
    const unfollowIcons = screen.getAllByAltText('Unfollow icon');
    try {
      await act(async () => {
        await userEvent.click(unfollowIcons[0]);
      });
      expect(unfollowIcons[0]).not.toBeInTheDocument();
      expect(unfollowIcons[1]).toBeInTheDocument();
      const followedUser = screen.getByText(/Alberto Contador/);
      expect(followedUser).toBeInTheDocument();
    } catch (error) {
      console.log(error);
    }
  });
  test('Test 5: User removes a follower', async () => {
    render(
      <BrowserRouter>
        <Follows userProfile={testUser} />
      </BrowserRouter>,
    );
    const followersCopy = [...testUser.followers];
    const removedFollower = followersCopy.splice(1, 1);
    mockAxios.onPost('/account/unfollow', { removedFollowing: removedFollower[0], newFollowList: followersCopy })
      .reply(200);
    const removeFollowerIcons = screen.getAllByAltText('Remove icon');
    try {
      await act(async () => {
        await userEvent.click(removeFollowerIcons[1]);
      });
      expect(removeFollowerIcons[1]).not.toBeInTheDocument();
      expect(removeFollowerIcons[0]).toBeInTheDocument();
      expect(removeFollowerIcons[2]).toBeInTheDocument();
      let followedUser = screen.getByText(/Cadel Evans/);
      expect(followedUser).toBeInTheDocument();
      followedUser = screen.getByText(/Alex Dowsett/);
      expect(followedUser).toBeInTheDocument();
    } catch (error) {
      console.log(error);
    }
  });
  test('Test 6: User reviews someone they follow', async () => {
    const userToReview = {
      name: 'Cadel Evans',
      email: 'cevans@sas.upenn.edu',
      reviews: [],
    };
    render(
      <BrowserRouter>
        <Follows userProfile={testUser} />
      </BrowserRouter>,
    );
    mockAxios.onPost('/account/findUserOnEmail', { email: 'cevans@sas.upenn.edu' }).reply(200, [userToReview]);
    const reviewIcons = screen.getAllByAltText('Review icon');
    try {
      await act(async () => {
        await userEvent.click(reviewIcons[0]);
      });
      const ratingInput = screen.getByRole('combobox');
      ratingInput.value = 5;
      const ratingContent = screen.getByRole('textbox');
      ratingContent.value = 'Testing review.';
      let modalHeaderText = screen.getByText(/Write a Review for Cadel Evans/);
      expect(modalHeaderText).toBeVisible();
      const reviewBtn = screen.getByText(/Submit/);
      mockAxios.onPost('/account/postReview', {
        author: testUser, recipient: userToReview, reviewRating: '5', reviewContent: 'Testing review.',
      }).reply(200);
      await act(async () => {
        await userEvent.click(reviewBtn);
      });
      modalHeaderText = screen.queryByText(/Write a Review for Cadel Evans/);
      expect(modalHeaderText).not.toBeVisible();
    } catch (error) {
      console.log(error);
    }
  });
  test('Test 7: User attempts to review someone they have already reviewed', async () => {
    const userToReview = {
      name: 'Cadel Evans',
      email: 'cevans@sas.upenn.edu',
      reviews: [
        {
          authorEmail: 'jdoe@sas.upenn.edu',
          authorName: 'John Doe',
          recipient: 'cevans@sas.upenn.edu',
          reviewContent: 'Good experience with this seller.',
          reviewRating: 5,
        },
      ],
    };
    render(
      <BrowserRouter>
        <Follows userProfile={testUser} />
      </BrowserRouter>,
    );
    mockAxios.onPost('/account/findUserOnEmail', { email: 'cevans@sas.upenn.edu' }).reply(200, [userToReview]);
    const reviewIcons = screen.getAllByAltText('Review icon');
    try {
      await act(async () => {
        await userEvent.click(reviewIcons[0]);
      });
      const modalText = screen.getByText(/You have already posted a review for this user./);
      expect(modalText).toBeVisible();
    } catch (error) {
      console.log(error);
    }
  });
});

describe('UI testing for the blocked component', () => {
  test('Test 1: Snapshot test', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Blocked blocked={testUser.blocked} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Test 2: UI reacts correctly when blocking another user', async () => {
    render(
      <BrowserRouter>
        <Blocked blocked={testUser.blocked} />
      </BrowserRouter>,
    );
    const unblockBtn = screen.getByAltText(/Unblock Icon/);
    try {
      mockAxios.onPost('/account/unblock', { newBlockedUsers: [] }).reply(200);
      await act(async () => {
        await userEvent.click(unblockBtn);
      });
      const noBlockedText = screen.getByText(/You have no one blocked./);
      expect(noBlockedText).toBeInTheDocument();
    } catch (error) {
      console.log(error);
    }
  });
});

// Chat is disabled in this component to remove testing errors.
describe('UI testing for SearchUsers component', () => {
  let userOne;
  let userTwo;
  beforeEach(async () => {
    userOne = {
      name: 'Jens Voigt',
      email: 'jvoigt@sas.upenn.edu',
      reviews: [],
      followers: [],
      following: [],
      reports: [],
      blocked: [],
    };
    userTwo = {
      name: 'Levi Leipheimer',
      email: 'leipheimer@sas.upenn.edu',
      reviews: [],
      followers: [
        {
          followerName: 'John Doe',
          followerEmail: 'jdoe@sas.upenn.edu',
          followingName: 'Levi Leipheimer',
          followingEmail: 'leipheimer@sas.upenn.edu',
        },
      ],
      following: [],
      reports: [
        {
          authorEmail: 'jdoe@sas.upenn.edu',
          recipientEmail: 'leipheimer@sas.upenn.edu',
          reportContent: 'Testing report.',
        },
      ],
      blocked: [],
    };
    render(<BrowserRouter><SearchUsers userProfile={testUser} /></BrowserRouter>);
    const searchInput = screen.getByRole('textbox');
    searchInput.value = 'i';
    const searchBtn = screen.getByRole('button');
    try {
      mockAxios.onPost('/account/findUsersOnName', { name: 'i' }).reply(200, [userOne, userTwo]);
      await act(async () => {
        await userEvent.click(searchBtn);
      });
    } catch (error) {
      console.log(error);
    }
  });

  test('Test 1: User searches for other users', async () => {
    const userOneResult = screen.getByText(/Jens Voigt/);
    const userTwoResult = screen.getByText(/Levi Leipheimer/);
    expect(userOneResult).toBeVisible();
    expect(userTwoResult).toBeVisible();
  });
  test('Test 2: User reports another user', async () => {
    const reportBtns = screen.getAllByText('Report');
    try {
      mockAxios.onPost('/account/findUserOnEmail', { email: 'jvoigt@sas.upenn.edu' }).reply(200, [userOne]);
      await act(async () => {
        await userEvent.click(reportBtns[0]);
      });
      const reportPrompt = screen.getByText(/Why are you reporting this user/i);
      expect(reportPrompt).toBeVisible();
      const reportContent = screen.getAllByRole('textbox')[1];
      reportContent.value = 'Testing 2 report';
      const reportBtn = screen.getAllByText(/Report/)[2];
      mockAxios.onPost('/account/postReport', { recipient: userOne, reportContent: reportContent.value }).reply(200);
      await act(async () => {
        await userEvent.click(reportBtn);
      });
      expect(reportPrompt).not.toBeInTheDocument();
    } catch (error) {
      console.log(error);
    }
  });
  test('Test 2: User reports someone they already reported', async () => {
    const reportBtns = screen.getAllByText('Report');
    try {
      mockAxios.onPost('/account/findUserOnEmail', { email: 'leipheimer@sas.upenn.edu' }).reply(200, [userTwo]);
      await act(async () => {
        await userEvent.click(reportBtns[1]);
      });
      const message = screen.getByText(/You have already reported Levi Leipheimer/);
      expect(message).toBeVisible();
    } catch (error) {
      console.log(error);
    }
  });
  test('Test 3: User follows another user', async () => {
    const followBtns = screen.getAllByText('Follow');
    try {
      mockAxios.onPost('/account/follow', { follower: testUser, followedUser: userOne }).reply(200);
      await act(async () => {
        await userEvent.click(followBtns[0]);
      });
      const message = screen.getByText(/You are now following Jens Voigt./);
      expect(message).toBeVisible();
    } catch (error) {
      console.log(error);
    }
  });
  test('Test 4: User attemps to follow an already followed user', async () => {
    const followBtns = screen.getAllByText('Follow');
    try {
      mockAxios.onPost('/account/follow', { follower: testUser, followedUser: userTwo }).reply(200);
      await act(async () => {
        await userEvent.click(followBtns[1]);
      });
      const message = screen.getByText(/You are already following Levi Leipheimer./);
      expect(message).toBeVisible();
    } catch (error) {
      console.log(error);
    }
  });
  test('Test 5: User blocks another user', async () => {
    const blockBtns = screen.getAllByText('Block');
    try {
      mockAxios.onPost('/account/block', { blocker: testUser, blockedUser: userOne }).reply(200);
      await act(async () => {
        await userEvent.click(blockBtns[0]);
      });
      const message = screen.getByText(/Jens Voigt is now blocked and you will no longer receive any messages or listings from them./);
      expect(message).toBeVisible();
    } catch (error) {
      console.log(error);
    }
  });
  test('Test 4: User attemps to follow an already followed user', async () => {
    const blockBtns = screen.getAllByText('Block');
    try {
      mockAxios.onPost('/account/block', { blocker: testUser, blockedUser: userTwo }).reply(200);
      await act(async () => {
        await userEvent.click(blockBtns[1]);
      });
      const message = screen.getByText(/You have already blocked Levi Leipheimer./);
      expect(message).toBeVisible();
    } catch (error) {
      console.log(error);
    }
  });
});
