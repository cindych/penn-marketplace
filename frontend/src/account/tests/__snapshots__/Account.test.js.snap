// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI testing for the Profile component Test 1: Snapshot on Profile Screen for user with a report 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="flex"
  >
    <h1>
      Profile
    </h1>
    <div
      className="flex"
      id="reports"
    >
      <img
        alt="Exclamation mark symbolizing warning."
        height="18px"
        src={
          Object {
            "process": [Function],
          }
        }
        width="18px"
      />
       
      1
       other users have reported you.
    </div>
  </div>
  <div
    id="profile"
  >
    <div
      style={
        Object {
          "padding": "1% 2%",
        }
      }
    >
      <div
        className="flex"
      >
        <p>
          Name
        </p>
        <p>
          John Doe
        </p>
      </div>
      <div
        className="flex"
      >
        <p>
          Email
        </p>
        <p>
          jdoe@sas.upenn.edu
        </p>
      </div>
      <div
        className="flex"
      >
        <p>
          Dark Mode
        </p>
        <button
          id="dark-mode-btn"
          type="button"
        >
          <img
            alt="Dark mode icon: off"
            src={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "verticalAlign": "top",
              }
            }
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`UI testing for the Profile component Test 2: Snapshot on Profile screen for user with no reports 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="flex"
  >
    <h1>
      Profile
    </h1>
    <div
      className="flex"
      id="reports"
    />
  </div>
  <div
    id="profile"
  >
    <div
      style={
        Object {
          "padding": "1% 2%",
        }
      }
    >
      <div
        className="flex"
      >
        <p>
          Name
        </p>
        <p>
          John Doe
        </p>
      </div>
      <div
        className="flex"
      >
        <p>
          Email
        </p>
        <p>
          jdoe@sas.upenn.edu
        </p>
      </div>
      <div
        className="flex"
      >
        <p>
          Dark Mode
        </p>
        <button
          id="dark-mode-btn"
          type="button"
        >
          <img
            alt="Dark mode icon: off"
            src={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "verticalAlign": "top",
              }
            }
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`UI testing for the blocked component Test 1: Snapshot test 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div>
    <div
      className="flex"
    >
      <h1>
        Blocked
      </h1>
      <div
        className="flex pagination-bar"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Back Arrow Icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </button>
         
        1
         - 
        1
          of 
        1
         
        <button
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Next Arrow Icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </button>
      </div>
    </div>
    <div
      className="box"
    >
      <div
        style={
          Object {
            "padding": "1% 2%",
          }
        }
      >
        <div
          className="table-row"
        >
          <br />
          <div>
            <br />
          </div>
          <div
            className="table-item"
          >
            Unblock
          </div>
        </div>
        <div
          className="table-row"
        >
          <p>
            Levi Leipheimer
          </p>
          <div>
            <br />
          </div>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={0}
            >
              <img
                alt="Unblock Icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UI testing for the follows component Test 1: Snapshot on Review screen for user with followers and followed users 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div>
    <div
      className="flex"
    >
      <h1>
        Followed
      </h1>
      <div
        className="flex pagination-bar"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Back Arrow Icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </button>
         
        1
         - 
        3
          of 
        3
         
        <button
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Next Arrow Icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </button>
      </div>
    </div>
    <div
      className="box"
    >
      <div
        style={
          Object {
            "padding": "1% 2%",
          }
        }
      >
        <div
          className="table-row"
        >
          <br />
          <div
            className="table-item"
          >
            Review
          </div>
          <div
            className="table-item"
          >
            Unfollow
          </div>
        </div>
        <div
          className="table-row"
        >
          <p>
            Cadel Evans
          </p>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={0}
            >
              <img
                alt="Review icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={0}
            >
              <img
                alt="Unfollow icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
        </div>
        <div
          className="table-row"
        >
          <p>
            Alberto Contador
          </p>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={1}
            >
              <img
                alt="Review icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={1}
            >
              <img
                alt="Unfollow icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
        </div>
        <div
          className="table-row"
        >
          <p>
            Levi Leipheimer
          </p>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={2}
            >
              <img
                alt="Review icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={2}
            >
              <img
                alt="Unfollow icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "3%",
      }
    }
  >
    <div
      className="flex"
    >
      <h1>
        Followers
      </h1>
      <div
        className="flex pagination-bar"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Back Arrow Icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </button>
        3
          of 
        3
         
        <button
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Next Arrow Icon"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </button>
      </div>
    </div>
    <div
      className="box"
    >
      <div
        style={
          Object {
            "padding": "1% 2%",
          }
        }
      >
        <div
          className="table-row"
        >
          <br />
          <div>
            <br />
          </div>
          <div
            className="table-item"
          >
            Remove
          </div>
        </div>
        <div
          className="table-row"
        >
          <p>
            Cadel Evans
          </p>
          <div>
            <br />
          </div>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={0}
            >
              <img
                alt="Remove icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
        </div>
        <div
          className="table-row"
        >
          <p>
            Frank Schleck
          </p>
          <div>
            <br />
          </div>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={1}
            >
              <img
                alt="Remove icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
        </div>
        <div
          className="table-row"
        >
          <p>
            Alex Dowsett
          </p>
          <div>
            <br />
          </div>
          <div
            className="table-item"
          >
            <button
              onClick={[Function]}
              type="button"
              value={2}
            >
              <img
                alt="Remove icon"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UI testing for the reviews component Test 1: Snapshot on Review Screen for user with reviews 1`] = `
<div>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <h1>
      Reviews - Average Rating: 
      3.5
    </h1>
  </div>
  <div
    id="reviews"
  >
    <div
      style={
        Object {
          "padding": "1% 2%",
        }
      }
    >
      <div
        className="review"
      >
        <div
          style={
            Object {
              "margin": "1%",
            }
          }
        >
          <div
            className="flex"
          >
            <div
              className="rating"
            >
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </div>
             
            Cadel Evans
          </div>
          <div>
            Good experience with this seller.
          </div>
        </div>
      </div>
      <div
        className="review"
      >
        <div
          style={
            Object {
              "margin": "1%",
            }
          }
        >
          <div
            className="flex"
          >
            <div
              className="rating"
            >
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <img
                alt="Yellow star."
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </div>
             
            Frank Schleck
          </div>
          <div>
            The chair I bought had a missing leg.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UI testing for the reviews component Test 2: Snapshot on Review screen for user with no reviews 1`] = `
<div>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <h1>
      Reviews - Average Rating: 
      0
    </h1>
  </div>
  <div>
    You have no reviews.
  </div>
</div>
`;
